File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> m;
    for(auto c : clothes) m[c[1]]++;
    int answer = 1;
    for(auto x: m) answer *= (x.second + 1);    
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> closet = new HashMap<>();
        
        for(int i = 0; i < clothes.length; i++)
            closet.put(clothes[i][1], closet.getOrDefault(clothes[i][1], 0) + 1);
        
        for(Map.Entry<String,Integer> entry : closet.entrySet()){
            answer *= (entry.getValue() + 1);
        }
        
        return answer - 1;
    }
}